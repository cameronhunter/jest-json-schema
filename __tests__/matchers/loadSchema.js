const createLoadSchema = require('../../matchers/loadSchema');
const path = require('path');
const fetch = require('node-fetch');
const remoteSchema = require('./__fixtures__/remote-schema.json');
const serializer = require('jest-serializer-path');

expect.addSnapshotSerializer(serializer);
jest.mock('node-fetch');

const fixtures = path.join(__dirname, '__fixtures__');

describe('loadSchema', () => {
  it('resolves local files', () => {
    const loadSchema = createLoadSchema({ 'http://my-domain.com/schema': fixtures });

    return expect(
      loadSchema('http://my-domain.com/schema/schema-with-defs.json')
    ).resolves.toMatchSnapshot();
  });

  it('rejects local files that cannot be loaded', () => {
    const loadSchema = createLoadSchema({ 'http://my-domain.com/schema': fixtures });

    return expect(
      loadSchema('http://my-domain.com/schema/nonexistent-schema.json')
    ).rejects.toThrowErrorMatchingSnapshot();
  });

  it('resolves remote files', () => {
    const loadSchema = createLoadSchema();

    fetch.mockImplementation(() => Promise.resolve({ ok: true, json: () => remoteSchema }));

    return expect(
      loadSchema('http://not-my-domain.com/remote-schema.json')
    ).resolves.toMatchSnapshot();
  });

  it('rejects if remote files cannot be fetched', () => {
    const loadSchema = createLoadSchema();

    fetch.mockImplementation(() => Promise.resolve({ ok: false }));

    return expect(
      loadSchema('http://not-my-domain.com/remote-schema.json')
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
