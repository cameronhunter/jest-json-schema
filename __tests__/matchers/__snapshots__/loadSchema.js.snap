// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadSchema rejects if remote files cannot be fetched 1`] = `"Couldn't fetch referenced schema \\"http://not-my-domain.com/remote-schema.json\\"."`;

exports[`loadSchema rejects local files that cannot be loaded 1`] = `"Cannot find module '<PROJECT_ROOT>/__tests__/matchers/__fixtures__/nonexistent-schema.json' from 'loadSchema.js'"`;

exports[`loadSchema resolves local files 1`] = `
Object {
  "$id": "http://example.com/schemas/schema-with-defs.json",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": Object {
    "int": Object {
      "type": "integer",
    },
    "str": Object {
      "type": "string",
    },
  },
}
`;

exports[`loadSchema resolves remote files 1`] = `
Object {
  "$id": "http://not-my-domain.com/remote-schema.json",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "properties": Object {
    "name": Object {
      "type": "string",
    },
  },
  "type": "object",
}
`;

exports[`loadSchema throws if there is no rootDir specified 1`] = `"Must define \\"rootDir\\" option"`;

exports[`loadSchema throws if there is no rootURI specified 1`] = `"Must define \\"rootURI\\" option"`;
